rfc/helm-synthesis.md

```diff
@@ -0,0 +1,206 @@
+# HELM CHART SYNTHESIS
+
+* **Original Author(s):**: @vinayak-kukreja
+* **Tracking Issue**: TODO:
```

Comment:
Will be creating issue for the RFC once internal review is done. Currently description links lead to nowhere.